### ISSUE TITLE
Update subscriptions similarly to the Targets Discovery mechanism

### ISSUE THREAD

**Issue by zerogvt** [openconfig/gnmic#293]
Hello :wave: 

gNMIc is a great project! Thanks a mil for sharing and supporting!

I'm in the process of adopting it as part of a statistics NMI pipeline. So far it works great and it is very promising. Only thing that I'm not I get right is the next use case:

Say that subscriptions need to change over the course of time. E.g. imagine I need to get stats on dropped packets on installed filters. Since these filters might come and go I'll need to keep updating my subscriptions which need to be in sync with the filter names. It would be great if that could be done via a mechanism similar to the targets discovery, i.e. if gNMIc could get these updates and fix the subscriptions on the fly.

Atm it seems I'll need to get the subscription updates from my backend, insert them in the configuration file and restart the gNMIc cluster which can be done but it is a bit cumbersome and deviates from the targets discovery workflow.

It'd be great if you got it into the roadmap somewhere if you see value in it.

Anyhow, thanks again!

Best

**Comment by Bapths** [comment]
Hi!

I know this have been posted a long time ago but I still think this could be really useful. When you have lots of different targets and you want to add new subscriptions it can be really annoying to restart the whole thing for such a small change. Maybe managing this a similar way to targets discovery could be one solution. Having access to the API to add/remove/edit subscriptions could be another one?

Thanks for everyone working hard on this amazing tool ;)

**Comment by vuthinh0301** [comment]
I totally agree with this proposal. In large-scale production environments, having to restart the entire gnmic instance just to modify a few subscriptions introduces a lot of operational risks and can disrupt telemetry data collection for other targets.

A dynamic subscription management mechanism (via API or a discovery-like behavior) would be a massive improvement for day-2 operations. Hoping to see this feature considered soon! Thanks for the amazing tool!
